include/validator/validator_component.h

```diff
@@ -87,6 +137,63 @@ struct ExternDescVisitor {
   Expect<void> operator()(const ValueType &) { return {}; }
 };
 
+struct CoreInstanceExprVisitor {
+  CoreInstanceExprVisitor(Context &Ctx) : Ctx(Ctx) {}
+
+  Expect<void> operator()(const CoreInstantiate &Inst) {
+    const uint32_t ModuleIdx = Inst.getModuleIdx();
+    if (ModuleIdx >= Ctx.getCoreModuleCount()) {
+      spdlog::error(ErrCode::Value::InvalidIndex);
+      spdlog::error(
+          "CoreInstanceSection: Module index {} exceeds available core modules {}"sv,
+          ModuleIdx, Ctx.getCoreModuleCount());
+      return Unexpect(ErrCode::Value::InvalidIndex);
+    }
+    auto Args = Inst.getArgs();
+    auto ImportsList = getImports(ModuleIdx);
+    for (auto import : ImportsList) {
+      auto argIt = std::find_if(Args.begin(), Args.end(), [&](const auto &Arg) {
+        return Arg.getName() == import;
+      });
+
+      if (argIt == Args.end()) {
+        spdlog::error(ErrCode::Value::MissingArgument);
+        spdlog::error("Module[{}]: Missing argument for import '{}'"sv,
+                      Inst.getModuleIdx(), import);
+        return Unexpect(ErrCode::Value::MissingArgument);
+      }
+    }
+    return {};
+  }
+
+  Expect<void> operator()(const CoreInlineExports &) { return {}; }
+
+private:
+  Context &Ctx;
+
+  std::vector<std::string> getImports(uint32_t Index) {
+    std::vector<std::string> ImportsList;
+
+    if (Index >= Ctx.getCoreModuleCount()) {
+      spdlog::error(
+          "Unreachable State: Index {} exceeds Core Module Count {}"sv, Index,
+          Ctx.getCoreModuleCount());
+      spdlog::error(WasmEdge::ErrInfo::InfoBoundary(
+          Index, 0, static_cast<uint32_t>(Ctx.getCoreModuleCount())));
+
+      return ImportsList;
+    }
```

Comment:
Here repeat the check above, and not returns an error. Maybe just remove this block.